cunumeric/array.py

```diff
@@ -3805,3 +3835,69 @@ def _perform_where(cls, mask, one, two) -> ndarray:
         out = ndarray(shape=out_shape, dtype=common_type, inputs=args)
         out._thunk.where(mask, one, two)
         return out
+
+    @classmethod
+    def _perform_scan(
+        cls, op, src, axis=None, dtype=None, out=None, nan0=False
+    ) -> ndarray:
+        if dtype is None:
+            if out is None:
+                if np.issubdtype(src.dtype, np.integer):
+                    # Set dtype to default platform integer
+                    dtype = np.int_
+                else:
+                    dtype = src.dtype
+            else:
+                dtype = out.dtype
+        if (
+            np.issubdtype(src.dtype, np.floating) or np.iscomplexobj(src)
+        ) and np.issubdtype(dtype, np.integer):
+            # Needs changes to convert()
+            raise NotImplementedError(
+                "Integer output types currently not supported for "
+                "floating/complex inputs"
+            )
+        if np.issubdtype(dtype, np.complexfloating) and nan0 is True:
+            # Currently not behaving correctly
+            raise NotImplementedError(
+                "nancumsum and nancumprod currently do not "
+                "supported complex types"
+            )
+        if out is not None:
+            if dtype != out.dtype:
+                # if out array is specified, its type overrules dtype
+                dtype = out.dtype
+            if axis is not None:
+                if out.shape != src.shape:
+                    raise RuntimeError(
```

Comment:
should use `NotImplementedError` instead

Reply:
24a420bcf34396f19e4fa2450a18975e658387d1